app/code/Magento/Widget/Model/Config/Converter.php

```diff
@@ -34,56 +49,99 @@ public function convert($source)
-                        break;
-                    case '#comment':
-                        break;
-                    default:
-                        throw new \LogicException(
-                            sprintf(
-                                "Unsupported child xml node '%s' found in the 'widget' node",
-                                $widgetSubNode->nodeName
-                            )
-                        );
-                }
+                $widgetArray = $this->processWidgetSubNode($widgetSubNode, $widgetArray);
             }
             $widgets[$widgetId->nodeValue] = $widgetArray;
         }
         return $widgets;
     }
 
+    /**
+     * Convert dom sub node to Magento array
+     *
+     * @param \DOMNode $widgetSubNode
+     * @param array $widgetArray
+     * @return array
+     * @throws \LogicException
+     */
+    protected function processWidgetSubNode($widgetSubNode, $widgetArray)
+    {
+        switch ($widgetSubNode->nodeName) {
+            case 'label':
+                $widgetArray['name'] = $widgetSubNode->nodeValue;
+                break;
+            case 'description':
+                $widgetArray['description'] = $widgetSubNode->nodeValue;
+                break;
+            case 'parameters':
+                $widgetArray = $this->processParameters($widgetSubNode, $widgetArray);
+                break;
+            case 'containers':
+                $widgetArray = $this->processContainers($widgetSubNode, $widgetArray);
+                break;
+            case "#text":
+                break;
+            case '#comment':
+                break;
+            default:
+                throw new \LogicException(
+                    sprintf(
+                        "Unsupported child xml node '%s' found in the 'widget' node",
+                        $widgetSubNode->nodeName
+                    )
+                );
+        }
+
+        return $widgetArray;
+    }
+
+    /**
+     * Convert dom sub node to Magento array
+     *
+     * @param \DOMNode $widgetSubNode
+     * @param array $widgetArray
+     * @return array
+     */
+    protected function processParameters($widgetSubNode, $widgetArray)
+    {
+        /** @var $parameter \DOMNode */
+        foreach ($widgetSubNode->childNodes as $parameter) {
+            if ($parameter->nodeName === '#text' || $parameter->nodeName === '#comment') {
+                continue;
+            }
+            $subNodeAttributes = $parameter->attributes;
+            $parameterName = $subNodeAttributes->getNamedItem('name')->nodeValue;
+            $widgetArray['parameters'][$parameterName] = $this->_convertParameter($parameter);
+        }
+
+        return $widgetArray;
+    }
+
+    /**
+     * Convert dom sub node to Magento array
+     *
+     * @param \DOMNode $widgetSubNode
+     * @param array $widgetArray
+     * @return array
+     */
+    protected function processContainers($widgetSubNode, $widgetArray)
```

Comment:
Please, make a method private. Add strict types to the method arguments

Reply:
Done with all 4 changes you have suggested. It is still showing me 2 errors.
1 is restricting use of array_merge function in a loop and another is showing some database error without any report. Your input on the same would be appreciable.